gsoc/admin.py

```diff
@@ -396,13 +397,23 @@ def email(self, obj):
         return obj.user.email
 
     def get_queryset(self, request):
-        return UserProfile.all_objects.all()
+        """
+        This queryset is used to return UserProfile object based on the role of request.user
+        If the user is a super user then all users are returned.
+        If the user is a suborg admin then mentors and students of that suborg are returned.
+        """
+        user = request.user.userprofile_set.get()
+        if request.user.is_superuser:
+            return UserProfile.all_objects.all()
+        if user.role == 1:
```

Comment:
so a user can have more then one role, so you will need to check all roles for the current year

Reply:
if the user is superuser then we return all the UserProfiles sorted by latest year.
```
UserProfile.all_objects.all().order_by("gsoc_year")
```
else we check if any of the UserProfiles has role set as suborg admin
```
user_profiles = user.userprofile_set.filter(role=1).all()
```